crates/uv-python/src/installation.rs

```diff
@@ -399,18 +399,23 @@ impl PythonInstallation {
         });
 
         if has_stable_download {
-            let minor_version = format!(
-                "{}.{}{}",
-                interpreter.python_major(),
-                interpreter.python_minor(),
-                interpreter.variant().display_suffix()
-            );
-
-            warn_user!(
-                "You're using a pre-release version of Python ({}) but a stable version is available. Use `uv python upgrade {}` to upgrade.",
-                version,
-                minor_version
-            );
+            if let Some(upgrade_request) = download_request
+                .clone()
```

Comment:
Why do we need to clone it? Isn't this the last use of the type?

Reply:
Ah it was the iterator from the outer `download_request.iter_downloads(...)` call that held onto a reference. I've put this into the `has_stable_download` block instead to fix it.